pretext/project/__init__.py

```diff
@@ -341,25 +338,60 @@ def build(
                 )
 
             log.info(f"Preparing to build into {self.output_dir_abspath()}.")
-
-            build.build(
-                self.format,
-                self.source_abspath(),
-                self.publication_abspath(),
-                self.output_dir_abspath(),
-                self.output_filename,
-                self.stringparams,
-                custom_xsl=custom_xsl,
-                xmlid=xmlid,
-                zipped=self.compression is not None,
-                project_path=self._project.abspath(),
-                latex_engine=self.latex_engine,
-                executables=self._project._executables.dict(),
-                # TODO: what if this isn't defined? Should we have a default in the project file instead?
-                braille_mode=self.braille_mode,
-            )
+            if self.format == "html":
+                core.html(
+                    xml=self.source_abspath(),
+                    pub_file=self.publication_abspath().as_posix(),
+                    stringparams=self.stringparams,
+                    xmlid_root=xmlid,
+                    file_format=self.compression or "html",
+                    extra_xsl=custom_xsl,
+                    out_file=None,
+                    dest_dir=self.output_dir_abspath().as_posix(),
+                )
+                # For codechat:
+                project_path = utils.project_path(self.source_abspath())
+                assert (
+                    project_path is not None
+                ), f"Invalid project path to {self.source_abspath()}."
+                codechat.map_path_to_xml_id(
+                    self.source_abspath(), project_path, self.output_dir_abspath().as_posix()
```

Comment:
Drop `as_posix()` here -- it's not necessary.

Reply:
Something is strange here.  without the `.as_posix()` I get complaints: Argument of type "Path" cannot be assigned to parametry "des_dir" of type "str" in function "map_path_to_xml_id"